src/parts/Header/Header.js

```diff
@@ -1,9 +1,22 @@
+import { connect } from 'react-redux';
+import { logOutUserAsyc } from '../../store';
 import './header.scss';
 import { Navigation } from '../Navigation/';
 import { pages } from './constants';
 
-export const Header = ({ user, login, setLoginState }) => (
-  <header className="header">
-    <Navigation pages={pages} user={user} login={login} setLoginState={setLoginState} />
-  </header>
-);
+export class HeaderComponent extends Component {
+  setLoginState = () => {
+    this.props.dispatch(logOutUserAsyc());
+  }
+
+  render() {
+    const { user } = this.props;
+
+    return (
+      <header className="header">
+        <Navigation pages={pages} user={user} setLoginState={this.setLoginState} />
+      </header>
+    );
+  }
+}
+export const Header = connect(({ user }) => ({ user }))(HeaderComponent);
```

Comment:
I think, Header doesn't require to be connected to the store. But Navigation requires.